From 6570e7afe1eebd64022b871b9ebd9a76be267f57 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Sun, 29 Jun 2014 16:27:49 +0200
Subject: [PATCH] Be coherent in type usage

- no need to use off_t because it is read from strtoul()
- print unsigned int type correctly

When built with _FILE_OFFSET_BITS=64, off_t becomes an long long int
and was printed as an unsigned int which lead to a garbled output.

Signed-off-by: Colin Didier <colin.didier@devialet.com>
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 devmem2.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/devmem2.c b/devmem2.c
index 7732ecb..7aa5c79 100644
--- a/devmem2.c
+++ b/devmem2.c
@@ -58,8 +58,7 @@
 int main(int argc, char **argv) {
     int fd;
     void *map_base, *virt_addr; 
-	unsigned long read_result, writeval;
-	off_t target;
+	unsigned long target, read_result, writeval;
 	int access_type = 'w';
 	
 	if(argc < 2) {
@@ -101,7 +100,7 @@ int main(int argc, char **argv) {
 			fprintf(stderr, "Illegal data type '%c'.\n", access_type);
 			exit(2);
 	}
-    printf("Value at address 0x%X (%p): 0x%X\n", target, virt_addr, read_result); 
+    printf("Value at address 0x%lX (%p): 0x%X\n", target, virt_addr, read_result); 
     fflush(stdout);
 
 	if(argc > 3) {
@@ -120,7 +119,7 @@ int main(int argc, char **argv) {
 				read_result = *((unsigned long *) virt_addr);
 				break;
 		}
-		printf("Written 0x%X; readback 0x%X\n", writeval, read_result); 
+		printf("Written 0x%lX; readback 0x%X\n", writeval, read_result); 
 		fflush(stdout);
 	}
 	
-- 
2.0.0

